x-pack/plugins/security_solution/server/lists_integration/endpoint/handlers/exceptions_pre_update_handler.ts

```diff
@@ -15,16 +15,21 @@ import { EventFilterValidator, TrustedAppValidator } from '../validators';
 export const getExceptionsPreUpdateItemHandler = (
   endpointAppContextService: EndpointAppContextService
 ): ExceptionsListPreUpdateItemServerExtension['callback'] => {
-  return async function ({ data, context: { request } }): Promise<UpdateExceptionListItemOptions> {
-    const currentSavedItem = await endpointAppContextService
-      .getExceptionListsClient()
-      .getExceptionListItem({
-        id: data.id,
-        itemId: data.itemId,
-        namespaceType: data.namespaceType,
-      });
-
-    // We don't want to `throw` here becuase we don't know for sure that the item is one we care about.
+  return async function ({
+    data,
+    context: { request, exceptionListClient },
+  }): Promise<UpdateExceptionListItemOptions> {
+    if (data.namespaceType !== 'agnostic') {
+      return data;
+    }
+
+    const currentSavedItem = await exceptionListClient.getExceptionListItem({
+      id: data.id,
+      itemId: data.itemId,
+      namespaceType: data.namespaceType,
+    });
+
```

Comment:
Did you forget the validation here?

Reply:
@dasansol92 - No, this was just a refactor to remove use of `endpointAppContextService.getExceptionListsClient()` and replace it with using `exceptionListClient` from the `context` provided in the arguments.